sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/EmptyCollectionsShouldNotBeEnumerated.cs

```diff
@@ -101,9 +101,16 @@ private void CheckForEmptyCollectionAccess(CSharpExplodedGraph explodedGraph, Sy
             var emptyCollections = new HashSet<SyntaxNode>();
             var nonEmptyCollections = new HashSet<SyntaxNode>();
 
+            EventHandler explorationEnded =
+                (sender, args) => emptyCollections.Except(nonEmptyCollections)
+                    .Select(node => Diagnostic.Create(rule, node.GetLocation()))
+                    .ToList()
+                    .ForEach(d => context.ReportDiagnosticWhenActive(d));
+
             EventHandler<CollectionAccessedEventArgs> collectionAccessedHandler = (sender, args) =>
                 (args.IsEmpty ? emptyCollections : nonEmptyCollections).Add(args.Node);
 
+            explodedGraph.ExplorationEnded += explorationEnded;

```

Comment:
ExplorationEnded is not raised if the execution limit is reached. We will not report in complex methods, exactly as the "Condition evaluates to constant" rule does.

Reply:
That's not part of this PR/ticket but could you check the other rules to see if this is present where needed?

Reply:
I already checked. I think they are fine.